sqlgen/reflect.go

```diff
@@ -246,85 +223,25 @@ func (s *Schema) buildDescriptor(table string, primaryKeyType PrimaryKeyType, ty
 		return nil, fmt.Errorf("bad type %s: no primary key specified", typ)
 	}
 
-	scannables := &sync.Pool{
-		New: func() interface{} {
-			scannables := make([]interface{}, len(columns))
-			for i, column := range columns {
-				scannables[i] = column.Scannable()
-			}
-			return scannables
-		},
-	}
-
 	return &Table{
 		Name:           table,
 		Type:           typ,
 		PrimaryKeyType: primaryKeyType,
 
 		Columns:       columns,
 		ColumnsByName: columnsByName,
-
-		Scannables: scannables,
 	}, nil
 }
 
-type Scannable interface {
-	sql.Scanner
-	driver.Valuer
-}
-
 type Schema struct {
 	ByName map[string]*Table
 	ByType map[reflect.Type]*Table
-
-	scalarTypes map[reflect.Type]func() Scannable
 }
 
 func NewSchema() *Schema {
-	scalarTypes := make(map[reflect.Type]func() Scannable)
-	for typ, scannable := range defaultScannableTypes {
-		scalarTypes[typ] = scannable
-	}
-
 	return &Schema{
 		ByName: make(map[string]*Table),
 		ByType: make(map[reflect.Type]*Table),
-
-		scalarTypes: scalarTypes,
-	}
-}
-
-func (s *Schema) RegisterCustomScalar(scalar interface{}, makeScannable func() Scannable) error {
-	scalarTyp := reflect.TypeOf(scalar)
-	if scalarTyp.Kind() == reflect.Ptr {
-		return fmt.Errorf("scalar type %v must not be a pointer", scalarTyp)
-	}
-	if _, ok := s.scalarTypes[scalarTyp]; ok {
-		return fmt.Errorf("duplicate scalar type %v", scalarTyp)
-	}
-	s.scalarTypes[scalarTyp] = makeScannable
-	return nil
-}
-
-func (s *Schema) MustRegisterCustomScalar(scalar interface{}, makeScannable func() Scannable) {
-	if err := s.RegisterCustomScalar(scalar, makeScannable); err != nil {
-		panic(err)
-	}
-}
-
-func (s *Schema) RegisterSimpleScalar(scalar interface{}) error {
```

Comment:
The migration _should_ be as simple as deleting these calls everywhere. My strategy for migrating this would be deleting all the calls and checking to see if it works.